src/fountain.c

```diff
@@ -747,15 +751,15 @@ doforging(void)
     /* Mundane item fusions */
     if (!combi_done) {
         for (int i = 0; fusions[i][0] > 0; i++) {
-            if ((obj1->otyp == fusions[i][0] && obj2->otyp == fusions[i][1]) ||
-                (obj2->otyp == fusions[i][0] && obj1->otyp == fusions[i][1])) {
-                obj1->otyp = fusions[i][2];
+            if ((obj1->otyp == fusions[i][1] && obj2->otyp == fusions[i][2]) ||
+                (obj2->otyp == fusions[i][1] && obj1->otyp == fusions[i][2])) {
+                obj1->otyp = fusions[i][0];
                 break;
             }
         }
     }
     /* Take on the secondary object's material. */
-    if (valid_obj_material(obj1, obj2->material)) obj1->material = obj2->material;
+    // if (valid_obj_material(obj1, obj2->material)) obj1->material = obj2->material;
```

Comment:
whoops, this should be a separate commit. however, removing this seems to be necessary for making wand forging work correctly.